indico/modules/events/editing/controllers/base.py

```diff
@@ -7,30 +7,65 @@
 
 from __future__ import unicode_literals
 
+import re
+
 from flask import request, session
+from werkzeug.exceptions import Unauthorized
 
 from indico.modules.events.contributions.controllers.display import RHContributionDisplayBase
 from indico.modules.events.controllers.base import RHDisplayEventBase
 from indico.modules.events.editing.models.editable import Editable, EditableType
+from indico.modules.events.editing.settings import editing_settings
 from indico.modules.events.management.controllers.base import RHManageEventBase
 from indico.web.rh import RequireUserMixin
 
 
-class RHEditingBase(RequireUserMixin, RHDisplayEventBase):
+AUTH_BEARER_RE = re.compile(r"Bearer ([\w-]+)")
+
+
+class TokenAccessMixin(object):
+    SERVICE_ALLOWED = False
+
+    def _token_can_access(self):
+        auth_header = request.headers.get('Authorization')
+        if not self.SERVICE_ALLOWED or not auth_header:
+            return False
+
+        m = AUTH_BEARER_RE.match(auth_header)
```

Comment:
What about adding a `bearer_token` property to `IndicoRequest`, which returns either the token or None? Like this we avoid having to put this in any place where we want to use Bearer tokens (which I hope may be some API more endpoints in the future as well..).

In that case I'd allow anything in the token though, not just `[\w-]` - simply because IMO it should be up to the code checking the token if it wants to do any further checks beyond "does it match".